ckan/lib/datapreview.py

```diff
@@ -264,7 +264,9 @@ def add_views_to_resource(context,
             view = {'resource_id': resource_dict['id'],
                     'view_type': view_info['name'],
                     'title': view_info.get('default_title', _('View')),
-                    'description': view_info.get('default_description', '')}
+                    'title_fr': view_info.get('default_title', _('View')),
```

Comment:
this will set both title fields to "View" in the language of the user that created the resource. If we're doing this's it's better for us to hard-code english and french strings here.

Reply:
@wardi The `default_title` for the view plugins are defined with `p.toolkit._('Image')`. Would the translation interface be able to just automatically translate all of the default view titles? Of course in the database the `title_fr` would just be the same as English, which is what we do not want?

Do you think that we should add `default_title_fr` to all of the views plugins? (e.g. https://github.com/open-data/ckan/blob/canada-v2.9/ckanext/datatablesview/plugin.py#L54) and then instead of doing:
```
'title_fr': view_info.get('default_title', _('View')),
```
we can do:
```
'title_fr': view_info.get('default_title_fr', _('Voir')),
```

Reply:
@wardi ah okay. Yeah we would not really want the default English title to be in French either, I understand.

Is the schema we are modifying in the `resource_view_create_bilingual` method? (https://github.com/open-data/ckanext-canada/blob/canada-v2.9/ckanext/canada/plugins.py#L292)

Would I just need to modify the `data_dict` to include the missing `title_fr` and `description_fr`?

I can also see an issue in doing it in `resource_view_create_bilingual`, as it does not appear we would not have the view plugin info in that method. So at the time, the `data_dict` would contain the `title` already set to `view_info.get('default_title', _('View'))`

Reply:
@wardi hmmm turns out only some of the view plugins `title` does not use `_()`. But some of them do.

So not too sure if we are able to do the default views in our extension. Unless if we have a dirty looking dict for English and French titles right in the `resource_view_create_bilingual` method? Keyed on the `view_type` string.

E.g.
```
titles = {
   'image_view': {
      'en': 'Image',
      'fr': 'Image'
   },
   'audio_view': {
      'en': 'Audio',
      'fr': 'Audio'
   },
   'video_view': {
      'en': 'Video',
      'fr': 'Vidéo'
   },
   'text_view': {
      'en': 'Text',
      'fr': 'Texte'
   },
   'webpage_view': {
      'en': 'Website',
      'fr': 'Site d\'internet'
   },
   'recline_view': {
      'en': 'Data Explorer',
      'fr': ''
   },
   'recline_grid_view': {
      'en': 'Table',
      'fr': ''
   },
   'datatables_view': {
      'en': 'Table',
      'fr': ''
   }
}
```

Issue is that it is quite static and is not very well integrated with ckan and the feature of view plugins.

Reply:
I think that's fine, when a default view is added there will only be one.

Reply:
https://github.com/open-data/ckanext-canada/pull/1332 <- above discussion now referring to our plugin.